test/fw/ptl/utils/pbs_dshutils.py

```diff
@@ -280,20 +285,33 @@ def _set_file(self, hostname, fin, fout, append, vars):
             conf = self._parse_file(hostname, fin)
         else:
             conf = {}
-        conf = dict(conf.items() + vars.items())
+        conf = dict(conf.items() + variables.items())
+
+        if os.path.isfile(fout):
+            fout_stat = os.stat(fout)
+            user = fout_stat.st_uid
+            group = fout_stat.st_gid
+        else:
+            user = None
+            group = None
 
         try:
             fn = self.create_temp_file()
             self.chmod(path=fn, mode=0644)
             with open(fn, 'w') as fd:
                 for k, v in conf.items():
                     fd.write(str(k) + '=' + str(v) + '\n')
-            self.run_copy(hostname, fn, fout, level=logging.DEBUG2)
-            self.rm(path=fn)
+            rv = self.run_copy(hostname, fn, fout, uid=user, gid=group,
+                               level=logging.DEBUG2, sudo=sudo)
+            if rv['rc'] != 0:
+                raise PbsConfigError
```

Comment:
Would it be better to pass an error message to PbsConfigError saying that the copy failed?

Reply:
It doesn't matter.  This is just really to trigger the try/except.  The except block does have an error message.  If I put one here, it would just be ignored.